### PR TITLE
OS-58: Added Insuree

### PR DESCRIPTION
**Pull request by bartekkaminski**
Added two endpoints in Controller:
 - Get Insuree
 - Get Enquire Insuree

TASK:
https://openimis.atlassian.net/browse/OS-58

### CODE REVIEW

File: OpenImis.ModulesV2/InsureeModule/Models/GetEnquireModel.cs [hunk 1]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenImis.ModulesV2.InsureeModule.Models
+{
+    public class GetEnquireModel
```
Comment (dragos-dobre) [inline OpenImis.ModulesV2/InsureeModule/Models/GetEnquireModel.cs]: This is the format of GetInsureeModel

File: OpenImis.ModulesV2/InsureeModule/Models/GetInsureeModel.cs [hunk 2]
```diff
@@ -0,0 +1,23 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenImis.ModulesV2.InsureeModule.Models
+{
+    public class GetInsureeModel : GetEnquireModel
```
Comment (dragos-dobre) [inline OpenImis.ModulesV2/InsureeModule/Models/GetInsureeModel.cs]: This is the format of GetEnquireModel

File: OpenImis.ModulesV2/InsureeModule/Logic/InsureeLogic.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+        protected IInsureeRepository insureeRepository;
+
+        public InsureeLogic(IConfiguration configuration)
+        {
+            _configuration = configuration;
+
+            insureeRepository = new InsureeRepository(_configuration);
+        }
+
+        public GetInsureeModel Get(string chfid)
```
Comment (dragos-dobre) [inline OpenImis.ModulesV2/InsureeModule/Logic/InsureeLogic.cs]: This seems to be GetEnquire

File: OpenImis.ModulesV2/InsureeModule/Logic/InsureeLogic.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        {
+            GetInsureeModel response;
+
+            response = insureeRepository.Get(chfid);
+
+            return response;
+        }
+
+        public GetEnquireModel GetEnquire(string chfid)
```
Comment (dragos-dobre) [inline OpenImis.ModulesV2/InsureeModule/Logic/InsureeLogic.cs]: This seems to be Get

File: OpenImis.ModulesV2/InsureeModule/Repositories/InsureeRepository.cs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+    public class InsureeRepository : IInsureeRepository
+    {
+        private IConfiguration _configuration;
+
+        public InsureeRepository(IConfiguration configuration)
+        {
+            _configuration = configuration;
+        }
+
+        public GetInsureeModel Get(string chfid)
```
Comment (dragos-dobre) [inline OpenImis.ModulesV2/InsureeModule/Repositories/InsureeRepository.cs]: This is for GetEnquire as it calls the uspPolicyInquiry SP. We need another call for Get(chfid) that will only do a simple select and get the information in GetInsureeModel

File: OpenImis.RestApi/Controllers/V2/InsureeController.cs [hunk 6]
```diff
@@ -0,0 +1,75 @@
...
+                return NotFound();
+            }
+
+            return Ok(getInsureeModel);
+        }
+
+        [HasRights(Rights.InsureeEnquire)]
+        [HttpGet]
+        [Route("{chfid}/enquire")]
+        public IActionResult GetEnquire(string chfid)
```
Comment (dragos-dobre) [inline OpenImis.RestApi/Controllers/V2/InsureeController.cs]: Pay attention on the right call.

File: OpenImis.ModulesV2/InsureeModule/Models/GetEnquireModel.cs (hunk 1 above)
Comment (bartekkaminski) [reply OpenImis.ModulesV2/InsureeModule/Models/GetEnquireModel.cs]: Done

File: OpenImis.ModulesV2/InsureeModule/Models/GetInsureeModel.cs (hunk 2 above)
Comment (bartekkaminski) [reply OpenImis.ModulesV2/InsureeModule/Models/GetInsureeModel.cs]: Done

File: OpenImis.ModulesV2/InsureeModule/Logic/InsureeLogic.cs (hunk 3 above)
Comment (bartekkaminski) [reply OpenImis.ModulesV2/InsureeModule/Logic/InsureeLogic.cs]: Done

File: OpenImis.ModulesV2/InsureeModule/Logic/InsureeLogic.cs (hunk 4 above)
Comment (bartekkaminski) [reply OpenImis.ModulesV2/InsureeModule/Logic/InsureeLogic.cs]: Done

File: OpenImis.ModulesV2/InsureeModule/Repositories/InsureeRepository.cs (hunk 5 above)
Comment (bartekkaminski) [reply OpenImis.ModulesV2/InsureeModule/Repositories/InsureeRepository.cs]: Done

File: OpenImis.RestApi/Controllers/V2/InsureeController.cs (hunk 6 above)
Comment (bartekkaminski) [reply OpenImis.RestApi/Controllers/V2/InsureeController.cs]: Done

File: OpenImis.ModulesV2/InsureeModule/Models/DetailModel.cs [hunk 7]
```diff
@@ -0,0 +1,19 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace OpenImis.ModulesV2.InsureeModule.Models
+{
+    public class DetailModel
```
Comment (dragos-dobre) [inline OpenImis.ModulesV2/InsureeModule/Models/DetailModel.cs]: Change this to EnquireDetailModel.

**Comment by dragos-dobre** [comment]
Can you check the conflicts here?

**Pull request merged**